client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -394,7 +444,8 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
             .values()
             .asScala
             .flatMap(_.getAllMasterLocationsWithMinEpoch(shuffleId.toString).asScala)
-            .filter(_.getEpoch == 0)
+            .filter(p =>
+              (partitionType == PartitionType.REDUCE && p.getEpoch == 0) || (partitionType == PartitionType.MAP && p.getId == partitionId))
```

Comment:
Do we have a consumption that for MapPartition there will always be 1 epoch?

Reply:
No, but for map partition only one "partition-location-Id" for the map task.